request.js

```diff
@@ -921,7 +921,7 @@ Request.prototype.onRequestResponse = function (response) {
     })
 
     var responseContent
-    if (self.gzip) {
+    if (self.gzip && self.method !== 'HEAD') {
```

Comment:
Anyway still this PR fixes the mentioned issue in the description which is the most common use case.

@czardoz you can open up another issue and point to this one, so we can have a discussion there.